src/Tickets/Commerce/Order.php

```diff
@@ -1054,14 +1125,27 @@ public function is_order_locked( int $order_id ): bool {
 	 */
 	public function is_checkout_completed( int $order_id ): bool {
 		try {
-			// Direct query to overcome object cache.
-			return (bool) DB::get_var(
-				DB::prepare(
-					'SELECT meta_value FROM %i WHERE post_id = %d AND meta_key = %s',
-					DB::prefix( 'postmeta' ),
-					$order_id,
-					static::CHECKOUT_COMPLETED_META
-				)
+			/**
+			 * Filters whether the checkout is completed.
+			 *
+			 * Direct query to overcome object cache.
+			 *
+			 * @since TBD
+			 *
+			 * @param bool $is_completed Whether the checkout is completed.
+			 * @param int  $order_id     The order ID.
+			 */
+			return apply_filters(
```

Comment:
Personal preference is to do the query before the hook and then just pass the value to the hook.